go-controller/pkg/util/go_ovn.go

```diff
@@ -93,10 +96,63 @@ func initGoOvnSslClient(certFile, privKeyFile, caCertFile, address, db, serverNa
 	if err != nil {
 		return nil, fmt.Errorf("error creating SSL OVNDBClient for database %s at address %s: %s", db, address, err)
 	}
+	if err = updateSslKeyPair(db, certFile, privKeyFile, tlsConfig, ovndbclient); err != nil {
+		return nil, fmt.Errorf("error watching SSL OVNDBClient for database %s cert/key files: %s", db, err)
+	}
+
 	klog.Infof("Created OVNDB SSL client for db: %s", db)
 	return ovndbclient, nil
 }
 
+// Watch TLS key/cert files, and update the ovndb tlsConfig Certificate.
+// Call ovndbclient.Close() will disconnect underlying rpc2client connection.
+// With ovndbclient initalized with reconnect flag, rcp2client will reconnct with new tlsConfig Certificate.
+func updateSslKeyPair(ovndb, certFile, privKeyFile string, tlsConfig *tls.Config, ovndbclient goovn.Client) error {
+
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return err
+	}
+
+	go func() {
+		for {
+			select {
+			case event, ok := <-watcher.Events:
+				if ok && event.Op&(fsnotify.Write|fsnotify.Remove) != 0 {
+					cert, err := tls.LoadX509KeyPair(certFile, privKeyFile)
+					if err != nil {
+						klog.Infof("Cannot load new cert with cert %s key %s err %s", certFile, privKeyFile, err)
+						continue
+					}
+					if reflect.DeepEqual(tlsConfig.Certificates, []tls.Certificate{cert}) {
+						klog.Infof("TLS update already finished")
+						continue
+					}
+					tlsConfig.Certificates = []tls.Certificate{cert}
+					err = ovndbclient.Close()
```

Comment:
does this close gracefully? Wondering if not if we could be in trouble if the client was in the middle of a transaction.

Reply:
@trozet  thanks for reviewing my changes.
the API calls in rpc2.Close()
From the man page for rpc2 package.
https://pkg.go.dev/github.com/cenkalti/rpc2#Client.Close
Close waits for active calls to finish and closes the codec.